pkg/redirectpolicy/manager.go

```diff
@@ -0,0 +1,533 @@
+	var newBackends []backend
+	for _, currBk := range config.backends {
+		for _, removeBk := range backends {
+			if removeBk.StringWithProtocol() != currBk.StringWithProtocol() {
+				newBackends = append(newBackends, currBk)
+			}
+		}
+	}
+	config.backends = newBackends
+	rpm.notifyPolicyBackendDelete(config)
+}
+
+// Deletes service entry corresponding to the specified frontend from datapath.
+func (rpm *RedirectPolicyManager) notifyPolicyFrontendDelete(config *LRPConfig) {
+	if config.frontend == nil {
+		return
+	}
+	found, err := rpm.svcManager.DeleteService(*config.frontend)
+	delete(rpm.frontends, config.frontend.Hash())
+	config.frontend = nil
+	if !found || err != nil {
+		log.WithError(err).Debugf("Local redirect service for policy %s not deleted",
+			config.getPolicyID())
+	}
+}
+
+// Updates datapath with the deleted backend(s).
+func (rpm *RedirectPolicyManager) notifyPolicyBackendDelete(config *LRPConfig) {
+	if config.frontend.IP != nil {
+		if len(config.backends) > 0 {
+			rpm.upsertService(config)
+		} else {
+			// No backends so remove the service entry.
+			found, err := rpm.svcManager.DeleteService(*config.frontend)
+			if !found || err != nil {
+				log.WithError(err).Debugf("Local redirect service for policy %s not deleted",
+					config.getPolicyID())
+			}
+		}
+	}
+}
+
+// Checks if the given policy config is ready, and plumbs datapath with a service
+// entry created from the policy config.
+func (rpm *RedirectPolicyManager) notifyPolicyConfigUpdate(config *LRPConfig) {
+	if config.frontend.IP != nil && len(config.backends) > 0 {
+		// Policy config is ready.
+		rpm.upsertService(config)
+	}
+}
+
+// Upserts a service entry for the given policy config that's ready.
+func (rpm *RedirectPolicyManager) upsertService(config *LRPConfig) {
+	frontendAddr := lb.L3n4AddrID{
+		L3n4Addr: *config.frontend,
+		ID:       lb.ID(0),
+	}
+	var backendAddrs []lb.Backend
+	for _, be := range config.backends {
+		backendAddrs = append(backendAddrs, lb.Backend{
+			NodeName: nodeTypes.GetName(),
+			L3n4Addr: be,
+		})
+	}
+	p := &lb.SVC{
+		Name:          config.name + localRedirectSvcStr,
+		Namespace:     config.namespace,
+		Type:          lb.SVCTypeLocalRedirect,
+		Frontend:      frontendAddr,
+		Backends:      backendAddrs,
+		TrafficPolicy: lb.SVCTrafficPolicyCluster,
+	}
+
+	if _, _, err := rpm.svcManager.UpsertService(p); err != nil {
+		log.WithError(err).Error("Error while inserting service in LB map")
+	}
+}
+
+// Returns a slice of endpoint pods metadata that are selected by the given policy config.
+func (rpm *RedirectPolicyManager) getLocalPodsForPolicy(config *LRPConfig, epManager endpointManager) []*epPodInfo {
+	var retEps []*epPodInfo
+
+	for _, ep := range epManager.GetEndpoints() {
+		podMetadata := rpm.getEndpointPodMetadata(ep)
+		if podMetadata == nil {
+			// TODO Pod metadata isn't available yet. We need to get callbacks
+			// when the metadata is available. Create a pending request.
```

Comment:
Maybe add a warning log here until this is resolved? I'd much rather make sure that we're it's more clear in the logs when this case was hit and that may cause connectivity issues for LRP rather than just having it not work quietly.